### ISSUE TITLE
PSVersionInfo test failing due to SMA version

### ISSUE THREAD

**Issue by rjmholt** [PowerShell/PowerShell#15607]
It looks like a PSVersionTable test is failing in CI, possibly because of a change to the version of System.Management.Automation.dll:

https://github.com/PowerShell/PowerShell/blob/945a23234170bd5ebdace8b6d10b82754354f68f/test/powershell/Host/PSVersionTable.Tests.ps1#L6-L24

**Comment by iSazonov** [comment]
All works well locally.
I build with ci.psm module as our CIs do
```powershell
Import-Module .\tools\ci.psm1
Invoke-CIBuild
```
 and get
![image](https://user-images.githubusercontent.com/22290914/122599768-49f73700-d088-11eb-9c7e-e78ed8ed7025.png)

It say something was changed in last release process.

**Comment by rjmholt** [comment]
> All works well locally.

Yeah I just noticed this.

So where's the 8 coming from?

**Comment by iSazonov** [comment]
I built from master and get 
![image](https://user-images.githubusercontent.com/22290914/122603096-4ca85b00-d08d-11eb-8af4-b5a91856d3eb.png)

Previous build was from my PR branch. So the issue is only in my PR. :-(

**Comment by iSazonov** [comment]
> So where's the 8 coming from?

From ci.psm:
```powershell
function Get-ReleaseTag
{
    $metaDataPath = Join-Path -Path $PSScriptRoot -ChildPath 'metadata.json'
    $metaData = Get-Content $metaDataPath | ConvertFrom-Json
    $releaseTag = $metadata.NextReleaseTag
    if($env:BUILD_BUILID)
    {
        $releaseTag = $releaseTag.split('.')[0..2] -join '.'
        $releaseTag = $releaseTag + '.' + $env:BUILD_BUILID
    }
    return $releaseTag
}
```
NextReleaseTag is v7.2.0-preview.8"

**Comment by iSazonov** [comment]
>     if($env:BUILD_BUILID)

Oh, I wonder how this works currently and tests are passed if the env name should be BUILD_BUIL**D**ID. Came from #8822.

**Comment by iSazonov** [comment]
It turned out that ci.psm1 always assigns it directly to dotnet and the code under `if($env:BUILD_BUILID)` is dead code.

**Comment by iSazonov** [comment]
@TravisEz13 @adityapatwardhan Friendly ping.

**Comment by TravisEz13** [comment]
I don't see a clear action item expressed.

**Comment by iSazonov** [comment]
@TravisEz13 I don't understand your comment.

I guess `function Get-ReleaseTag` is broken and must be fixed. I don't know how since it is exclusively internal MSFT process. I'd want to get a fix commit so that #15603 passes the test.

**Comment by iSazonov** [comment]
I compared builds local, CI and release and _I am at a loss._

We always put Product version including GitCommitId on all files in all builds. It is right.
But we lost GitCommitId in banner and PSVersionTable in non-local builds! Did it happen by accident?

When we discussed adding a GitCommitId to the PSVersionTable, we found it important to inform users what the last commit was for this build. And we lost it. _It should be in the Release build_ and CI builds too.
(To be clear I expect short PSVersion in banner (like `PowerShell 7.2.0`) and full version with GitCommitId in PSVersionTable.GitCommitId (like `7.2.0-preview.10-166-g40837dd860e1bb69dd8238ff5e5ae360a7637c13`).

Perhaps this is a consequence of how the release process is going.
_This definitely violates generally accepted rules of open source projects._
Our Git Tags are broken. They seem to indicate the beginning of the next milestone, but they are not due to that we are backporting commits. The only way to do it right is to have _release branches in the repository_.

There is a related problem here. No community can release a version similar MSFT since nobody knows which commits were backported (specially in patches) and no public release branches exist - it breaks the open source nature of the project.
MSFT can have non-public release process but MSFT must keep release branches public with right git tags.

/cc @SteveL-MSFT @TravisEz13 @adityapatwardhan for review.
.
